### DISCUSSION TITLE
[help] Targets invalidated because of .Random.seed

### DISCUSSION THREAD

**Discussion by mireia-bioinfo** [ropensci/targets#1030] [Help]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I am currently working on a targets project in two different environments: an HPC cluster and my local workstation. I keep track of the code with git and I rsync the _targets folder, preserving file info with -a.

However, every time I change from one environment to the other, all my targets are invalidated. I finally managed to find the root of the problem, which is the global object `.Random.seed`. This object is different in my different environments and even if I use set.seed, it’s still different.

I’m sure this is something very basic but I am no able to find any info on how to fix this. Can anyone help?

Thanks in advance!

**Comment by wlandau** [comment]
You could call set.seed() at the top of _targets.R to make sure the .Random.seed object is always the same for the pipeline. (But different versions of R may have different ways of setting seeds, so it may be best to make sure those versions are the same too.) Or, to trick the static code analyzer into ignoring the dependency, you could call eval(parse(text = ".Random.seed")) to reference the seed instead of just typing .Random.seed. Note: both workarounds are different from the seed argument of tar_option_set() and tar_cue().

**Comment by snowpong** [comment]
I have a similar issue. In my targets `meta` file the `.Random.seed` changes each time I run `targets::tar_make()` even though nothing has changed and all targets are skipped. I guess this is by design? Since I don't want my diff to change unless my pipeline has changed the only solution was to call `set.seed()` in my `_targets.R` as suggested here. But, is this recommended practice?
